crates/wasmtime/src/config.rs

```diff
@@ -2303,6 +2308,38 @@ impl PoolingAllocationConfig {
         self.config.limits.memory_pages = pages;
         self
     }
+
+    /// Configures whether memory protection keys (MPK) should be used for more
+    /// efficient layout of pool-allocated memories.
+    ///
+    /// When using the pooling allocator (see [`Config::allocation_strategy`],
+    /// [`InstanceAllocationStrategy::Pooling`]), memory protection keys can
+    /// reduce the total amount of allocated virtual memory by eliminating guard
+    /// regions between WebAssembly memories in the pool. It does so by
+    /// "coloring" memory regions with different memory keys and setting which
+    /// regions are accessible each time executions switches from host to guest
+    /// (or vice versa).
+    ///
+    /// MPK is only available on Linux (called `pku` there) and recent x86
+    /// systems; we check for MPK support at runtime by examining the `CPUID`
+    /// register. This configuration setting can be in three states:
```

Comment:
Can we export a `PoolingAllocationConfig::mpk_available() -> bool` static method? We have that internally anyways, and it seems like it would be useful to embedders.

Reply:
I think `wasmtime_runtime::mpk::is_available()` should be public already which is in the same crate?

Reply:
`wasmtime_runtime` is an implementation detail that isn't available to users of the public `wasmtime` API. So I am imagining that we would expose that functionality in the public `wasmtime` API, and as a static method on `PoolingAllocationConfig` seemed like a nice place but I am happy to bikeshed.

This is also something that can be added after the initial MPK support lands.